cfx_types/src/lib.rs

```diff
@@ -26,3 +27,39 @@ pub fn hexstr_to_h256(hex_str: &str) -> H256 {
 
     H256::from(bytes)
 }
+
+pub trait AddressUtil {
+    fn type_bits(&self) -> u8;
+
+    fn is_valid<T>(&self, builtin_map: &BTreeMap<Address, T>) -> bool;
+
+    fn is_contract(&self) -> bool;
+
+    fn is_payment(&self) -> bool;
+
+    fn is_builtin<T>(&self, builtin_map: &BTreeMap<Address, T>) -> bool;
+
+    fn converted_to_contract(&mut self);
```

Comment:
set_contract_type_bits?

You may want to do something similar in public_to_address.

I feel that you an address_util mod fits the usage better.